be1-go/hub/serveWs.go

```diff
@@ -37,20 +52,43 @@ func serveWs(socketType SocketType, h Hub, w http.ResponseWriter, r *http.Reques
 
 	switch socketType {
 	case ClientSocketType:
-		client := NewClientSocket(h, conn)
+		client := NewClientSocket(h, conn, wg)
 
-		go client.ReadPump()
-		go client.WritePump()
+		go client.ReadPump(ctx)
+		go client.WritePump(ctx)
 
 		// cleanup go routine that removes clients that forgot to unsubscribe
 		go func(c *ClientSocket, h Hub) {
 			c.Wait.Wait()
 			h.RemoveClientSocket(c)
 		}(client, h)
 	case WitnessSocketType:
-		witness := NewWitnessSocket(h, conn)
+		witness := NewWitnessSocket(h, conn, wg)
+
+		go witness.ReadPump(ctx)
+		go witness.WritePump(ctx)
+	}
+}
+
+func ShutDownServers(ctx context.Context, cancel context.CancelFunc, witnessSrv *http.Server, clientSrv *http.Server, wg *sync.WaitGroup) {
```

Comment:
I think it's better if this is in the cli package instead - this isn't exactly tied to the hub. A nit about naming but I'd prefer `ShutdownServers` instead :)

Reply:
We can also avoid passing in `cancel` and `wg` if this is invoked as a blocking function

Reply:
Indeed good idea! I have renamed it and made it a blocking call to avoid passing it cancel and wg.